File: frontend/src/types/article.ts
export type Summary = {
  headline: string;
  background: string;
};

export type CardSlide =
  | { type: "cover"; title: string; subtitle?: string }
  | { type: "context"; body: string }
  | { type: "point"; heading: string; body: string }
  | { type: "outro"; headline: string; cta?: string };

export type CardNews = {
  slides: CardSlide[];
};

export type Article = {
  _id: string;
  source_id: string;
  category: string;
  subcategory?: string | null;
  lang: string;
  title: string;
  url: string;
  image_url?: string | null;
  content?: string | null;
  summary?: Summary | null;
  cardnews?: CardNews | null;
  meta_data?: Record<string, unknown>;
  published_at?: string | null;
  collected_at: string;
};

export type ArticleListResponse = {
  items: Article[];
  next_cursor: string | null;
};

export type CryptoTicker = {
  symbol: string;
  price_usd: number | null;
  change_24h_pct: number | null;
};

File: backend/app/models/article.py
from datetime import datetime
from typing import Optional, Any
from pydantic import BaseModel, Field


class Summary(BaseModel):
    headline: str = Field(description="핵심 1줄 결론")
    background: str = Field(description="배경과 전망 2줄 요약")


class Article(BaseModel):
    id: str = Field(alias="_id")
    source_id: str
    category: str
    subcategory: Optional[str] = None
    lang: str = "en"
    title: str
    url: str
    image_url: Optional[str] = None
    content: Optional[str] = None
    summary: Optional[Summary] = None
    meta_data: dict[str, Any] = Field(default_factory=dict)
    published_at: Optional[datetime] = None
    collected_at: datetime

    model_config = {"populate_by_name": True}


class ArticleListResponse(BaseModel):
    items: list[Article]
    total: int
